lib/improver/nowcasting/pysteps_advection.py

```diff
@@ -0,0 +1,244 @@
+                [current_time], dtype=np.int64)
+            new_cube.coord('forecast_period').points = np.array(
+                [(i+1)*self.interval*60], dtype=np.int32)
+            forecast_cubes.append(new_cube)
+        return forecast_cubes
+
+    def _generate_forecast_cubes(self, all_forecasts):
+        """
+        Convert forecast arrays into IMPROVER output cubes with re-added
+        orographic enhancement
+
+        Args:
+            all_forecasts (np.ndarray):
+                Array of 2D forecast fields returned by extrapolation function
+
+        Returns:
+            forecast_cubes (list):
+                List of iris.cube.Cube instances containing forecasts at all
+                required lead times, and conforming to the IMPROVER metadata
+                standard.
+        """
+        # re-mask forecast data
+        mask = np.where(np.isfinite(all_forecasts), False, True)
+        all_forecasts = np.ma.MaskedArray(all_forecasts, mask=mask)
+
+        # generate list of forecast cubes
+        self._reformat_analysis_cube()
+        timestamped_cubes = self._set_up_output_cubes(all_forecasts)
+
+        # re-convert units and re-add orographic enhancement
+        forecast_cubes = []
+        for cube in timestamped_cubes:
+            cube.convert_units(self.required_units)
+            cube, = ApplyOrographicEnhancement("add").process(
+                cube, self.orogenh)
+            forecast_cubes.append(cube)
+        return forecast_cubes
+
+    def process(self, initial_cube, ucube, vcube, interval, max_lead_time,
+                orographic_enhancement):
+        """
+        Extrapolate the initial precipitation field using the velocities
+        provided to the required forecast lead times
+
+        Args:
+            initial_cube (iris.cube.Cube):
+                Cube of precipitation at initial time
+            ucube (iris.cube.Cube):
+                x-advection velocities
+            vcube (iris.cube.Cube):
+                y-advection velocities
+            interval (int):
+                Lead time interval, in minutes
+            max_lead_time (int):
+                Maximum lead time required, in minutes
+            orographic_enhancement (iris.cube.Cube):
+                Cube containing orographic enhancement fields at all required
+                lead times
+
+        Returns:
+            forecast_cubes (list):
+                List of extrapolated iris.cube.Cube instances at the required
+                lead times (including T+0 / analysis time)
+        """
+        # ensure input cube is suitable for advection
+        if 'rate' not in initial_cube.name():
+            msg = '{} is not a precipitation rate cube'
+            raise ValueError(msg.format(initial_cube.name()))
+        check_if_grid_is_equal_area(initial_cube)
+
+        self.analysis_cube = initial_cube.copy()
+        self.required_units = initial_cube.units
+        self.interval = interval
```

Comment:
This plugin currently lacks init and repr functions. It feels like the init function would be the right place for the interval and max_leadtime arguments (and calculating num_timesteps), though if there is a good reason to leave them here, please explain.

Reply:
Have now added an `__init__` function to set `interval` and `num_timesteps`.